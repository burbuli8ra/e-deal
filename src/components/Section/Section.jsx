import React from 'react';
import { useAppContext } from 'provider';
import Styled from './Section.style';

const Section = ({ children }) => {
  const [{ componentHeights: { header, footer } }] = useAppContext();

  return (
    <Styled.Main
      data-testid="main"
      headerHeight={header}
      footerHeight={footer}
    >
      {children}
    </Styled.Main>
  );
};

export default Section;