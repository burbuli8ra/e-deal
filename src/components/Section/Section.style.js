import styled from '@emotion/styled';
import { colors } from 'theme';

const Main = styled.main`
  align-items: center;
  background-color: ${colors.lightGrey};
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  min-height: calc(100% - ${({ headerHeight, footerHeight }) => headerHeight + footerHeight}px);
  padding: 20px;
`;
Main.displayName = 'Main';

export default {
  Main
};