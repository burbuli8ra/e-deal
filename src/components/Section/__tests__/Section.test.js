import React from 'react';
import { render } from '@testing-library/react';
import { AppProvider } from 'provider';
import Section from '../Section';

const setup = props => (
  <AppProvider>
    <Section {...props}>
      Section content
    </Section>
  </AppProvider>
);

describe('Section component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {};
    wrapper = render(setup(props));
  });

  test('should match snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});