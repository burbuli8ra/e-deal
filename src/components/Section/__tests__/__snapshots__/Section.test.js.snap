// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component should match snapshot 1`] = `
<DocumentFragment>
  <main
    class="css-7sbsyk"
    data-testid="main"
  >
    Section content
  </main>
</DocumentFragment>
`;
